### PR TITLE
[MM-20061] Add System Admin filter to both LDAP and SAML

### PR DESCRIPTION
**Pull request by hahmadia**
**Merge First**

#### Summary

Similar to the current guest filter, do the same but for System Admin.

#### Ticket Link

https://mattermost.atlassian.net/browse/MM-20061

### CODE REVIEW

File: model/user.go [hunk 1]
```diff
@@ -633,6 +633,10 @@ func (u *User) IsGuest() bool {
 	return IsInRole(u.Roles, SYSTEM_GUEST_ROLE_ID)
 }
 
+func (u *User) IsAdmin() bool {
```
Comment (mkraft) [inline model/user.go]:
```suggestion
func (u *User) IsSystemAdmin() bool {
```
Just wondering if you think we should differentiate it from channel admin and team admin by adding the word System to the method name?

Comment (hahmadia) [reply model/user.go]: Good catch! I've changed it.

**Comment by hahmadia** [comment]
@jespino 
I added invalidating cache to some functions you worked on previously. Could you just confirm that it's okay to have them there? If I don't add them, the roles don't update when logging in and it just retrieves it from cache so that's why I had add those.

File: model/config.go [hunk 2]
```diff
@@ -1927,15 +1938,17 @@ type SamlSettings struct {
...
-	FirstNameAttribute *string
-	LastNameAttribute  *string
-	EmailAttribute     *string
-	UsernameAttribute  *string
-	NicknameAttribute  *string
-	LocaleAttribute    *string
-	PositionAttribute  *string
+	IdAttribute          *string
+	GuestAttribute       *string
+	EnableAdminAttribute *bool
```
Comment (jespino) [inline model/config.go]: Is there any reason to need this `EnableAdminAttribute` setting here, when we don't have an equivalent one for `GuestAttribute`? Makes sense to default the `AdminAttribute` to blank and that means "disabled" and if there is any value, is enabled? Is not that behavior already the expected one?

Comment (hahmadia) [reply model/config.go]: @jespino 
Actually we do have one for GuestAttribute and GuestFilter. It's the `Enable` in the `GuestAccountSettings` in config.json. However, that has more of an impact than just turning the `GuestAttribute` and `GuestFilter` on and off.

Without the `EnableAdminAttribute` and `EnableAdminFilter`, the problem we run into is if the filter/attribute is blank, then all members who are admins will become members when they login even if we don't want that to happen.  If we disable the filter, then the admins will not be demoted on the next login. This was the reason we added the Enable.

**Pull request merged**